src/Core/Twig/Extension/RoleExtension.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Forumify\Core\Twig\Extension;
+
+use Forumify\Core\Entity\Role;
+use Forumify\Core\Entity\User;
+use Forumify\Core\Repository\RoleRepository;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+class RoleExtension extends AbstractExtension
+{
+    public function __construct(
+        private RoleRepository $roleRepository
+    ){}
+    public function getFilters(): array
+    {
+        return [
+            new TwigFilter('role_color', [$this, 'roleColor']),
+        ];
+    }
+    public function roleColor(?User $user): ?string
+    {
+        return $user
+            ?->getRoleEntities()
+            ?->filter(fn(Role $r) =>
+                $r->isShowOnUsername()
+                && strtolower($r->getColor() ?? '') !== '#000000'
+            )
+            ?->first()
+            ?->getColor();
```

Comment:
So yeah this is my fault. `first()` can return `false`, and then you can't chain `->getColor()`. You'll have to split it and like

```
$role = ...
    ?->first();

return $role ? $role->getColor() : null;
```